### DISCUSSION TITLE
getShopByOwnerUserId V3 endpoint throws a Forbidden result

### DISCUSSION THREAD

**Discussion by WilmerGuerrero** [etsy/open-api#753] [Q&A] [answered]
The doc https://developers.etsy.com/documentation/reference/#operation/getShopByOwnerUserId says that just the x-api-key is needed for requesting this endpoint, but when I try this out I got an 403 with the following error message: **No user was provided as part of the bearer token.** For being sure that I didn't missed anything I tried with other endpoints that just require userId like: findAllListingsActive and the response was as expected. Am I missing something for making this work?

Another question not necessarily related to this is: we are migrating our integration from V2 to V3 and before the __SELF__ parameter worked for referencing the shopId. But, when I tried using it the endpoint looked like was not mapping the string to its related integer shopId. Is no longer working this param?

**Comment by etsyachristensen** [comment, marked as answer]
@WilmerGuerrero First off, the `SELF` parameter is removed in V3. You have to have your shop id and user id for the endpoints that require them. After authenticating with the OAuth and after the user has granted your app consent to access it, your app can call the `getMe` endpoint (https://developers.etsy.com/documentation/reference#operation/getMe).

For the other endpoints, the `x-api-key` has to be sent as a header value. All of our endpoints will require the `x-api-key` value (except for pre-auth). Once you've done the OAuth process you should have a bearer token that you would pass into every call. The token is what helps our system determine who the owner of the token is.
